### DISCUSSION TITLE
Metadata Search Secrets

### DISCUSSION THREAD

**Discussion by dionrowney** [thycotic-ps/thycotic.secretserver#404] [Q&A : Module] [answered]
I am having trouble trying to determine how to query secrets with metadata.

I have secrets which have a metadata section and field boolean checkbox.

I need to use powershell to extract the secrets which have this value = true but am having troubles finding where to get the stored value from.

Is there any cmdlet that like Get-Secret | %{ Get-MetadataFieldValue -fieldname x -itemId $_.id = true ...} sort of thing?

I was able to write a report that produced this result but was hoping to use just powershell as other operations are required on the results anyways.  her is the SQL I used in the GUI:

```
SELECT	
		s.SecretName AS 'Secret Name'
		,s.SecretId AS 'SecretId'
		,md.ValueBit as 'isCritical'
		,s.secretPolicyId as 'PolicyID'
        ,sp.SecretPolicyName as 'PolicyName'
        ,LastSuccessfulPasswordChangeDate as 'LastSuccessfulPasswordChangeDate'
        ,s.AutoChangeOnExpiration as 'autochange'
	FROM
		tbSecret s WITH (NOLOCK)
	
	INNER JOIN tbMetadataItemData md WITH (NOLOCK)
		ON s.SecretId = md.ItemId
	INNER JOIN tbSecretPolicy sp WITH (NOLOCK)
		ON s.secretPolicyId = sp.SecretPolicyId        
	where md.ValueBit = 1
```
	
Any ideas?

I guess they best way to ask is How can i see the data value in the metadata using powershell.  the docs just are not getting me there and I need an example to start from.

thanks

**Comment by dionrowney** [comment, marked as answer]
solved

did this:

Find-TssSecret -TssSession $session | %{ Search-TssMetadata -TssSession $session -ItemId $_.SecretId -ItemType Secret }

**Comment by jagger** [comment]
That will do it, you are querying the full list of secrets first. If
possible filter anything you can before doing the second search to limit
resource usage
